### PR TITLE
test: making E-M e2e test work sans yaml

### PR DESCRIPTION
**Pull request by alyssawilk**
bazel test --config=remote --define=envoy_yaml=disabled  //test/common/integration:client_integration_test

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
`Verify/examples` fail is not real - i dont think retest will work - ive added a fix so safe to ignore/should go away on push

File: .github/workflows/mobile-compile_time_options.yml [hunk 1]
```diff
@@ -43,7 +43,22 @@ jobs:
...
+            $([ -z $GITHUB_TOKEN ] || echo "--config=remote-ci-macos") \
+            --config=ci \
+            --fat_apk_cpu=x86_64 \
+            --define=signal_trace=disabled \
+            --define=envoy_mobile_request_compression=disabled \
+            --define=envoy_enable_http_datagrams=disabled \
+            --define=google_grpc=disabled \
+            --define=envoy_yaml=disabled \
+            --@com_envoyproxy_protoc_gen_validate//bazel:template-flavor= \
+            //test/java/integration:android_engine_socket_tag_test \
```
Comment (KBaichoo) [inline .github/workflows/mobile-compile_time_options.yml:58]: Context on moving this test from the Kotlin build pipeline to cc_test?

File: test/config/utility.cc [hunk 2]
```diff
@@ -636,6 +641,12 @@ ConfigHelper::buildBaseListener(const std::string& name, const std::string& addr
                                 name, address, filter_chains),
                             listener);
   return listener;
+#else
+  UNREFERENCED_PARAMETER(name);
+  UNREFERENCED_PARAMETER(address);
+  UNREFERENCED_PARAMETER(filter_chains);
+  PANIC("YAML support compiled out");
```
Comment (KBaichoo) [inline test/config/utility.cc:648]: Nice error message, maybe the others can have this thrown in as well.

e.g. 
https://github.com/envoyproxy/envoy/pull/27634/files#diff-1cf1654ea1233e6793e9a924bbb6ec36beff732bb7ea2c0230d9cc74b87e55f1R83 

https://github.com/envoyproxy/envoy/pull/27634/files#diff-1cf1654ea1233e6793e9a924bbb6ec36beff732bb7ea2c0230d9cc74b87e55f1R189

etc.

Alternatively, what's the thinking about putting this if-def throughout the codebase vs in `loadFromYaml`?

**Review by KBaichoo — COMMENTED** [review]
/assign @KBaichoo 

/wait

**Comment by phlax** [comment]
> verify examples fail looks unrelated

yeah just a network transient i think - not sure if

/retest 

...works

but even if it does it wont show on the PR until it has completed (annoyingly - ill try and fix this) 

you can make it work by doing "retest all" (rather than "retest failed")

**Comment by phlax** [comment]
```console
retest-action failure: HttpError: This workflow run cannot be retried
```

looks like manual only until i figure out something better

**Review by KBaichoo — APPROVED** [review]
Thanks!

**Pull request merged**